src/bitcask.erl

```diff
@@ -404,11 +404,19 @@ fold(State, Fun, Acc0, MaxAge, MaxPut, SeeTombstonesP) ->
                     {ok, Files, FoldEpoch} ->
                         ExpiryTime = expiry_time(State#bc_state.opts),
                         SubFun = fun(K0,V,TStamp,{_FN,FTS,Offset,_Sz},Acc) ->
-                                         K = KT(K0),
-                                         case (TStamp < ExpiryTime) of
-                                             true ->
+                                         K = try
+                                                 KT(K0)
+                                             catch
+                                                 KeyTxErr ->
+                                                     {key_tx_error, {K0, KeyTxErr}}
+                                             end,
+                                         case {K, (TStamp < ExpiryTime)} of
+                                             {{key_tx_error, TxErr}, _} ->
+                                                 error_logger:error_msg("Error converting key ~p", [TxErr]),
                                                  Acc;
-                                             false ->
+                                             {_, true} ->
+                                                 Acc;
+                                             {_, false} ->
                                                  case bitcask_nifs:keydir_get(
```

Comment:
Holy indentation Batman! Not your problem to fix, but this is really difficult to follow as a n00b.